projects/plugins/jetpack/extensions/blocks/related-posts/test/edit.js

```diff
@@ -54,29 +55,59 @@ const posts = [
 	},
 ];
 
-describe( 'RelatedPostsEdit', () => {
-	const defaultAttributes = {
-		displayContext: false,
-		displayDate: false,
-		displayThumbnails: false,
-		postLayout: 'grid',
-		postsToShow: '2',
-	};
-
-	const setAttributes = jest.fn();
-	const defaultProps = {
-		attributes: defaultAttributes,
-		setAttributes,
-		clientId: 1,
-		posts: posts,
-		className: 'className',
+jest.mock( '@automattic/jetpack-shared-extension-utils', () => ( {
+	__esModule: true,
+	...jest.requireActual( '@automattic/jetpack-shared-extension-utils' ),
+	useModuleStatus: jest.fn().mockReturnValue( {
+		isModuleActive: true,
+		isLoadingModules: false,
+		isChangingStatus: false,
+		changeStatus: jest.fn(),
+	} ),
+} ) );
+jest.mock( '@wordpress/element', () => ( {
```

Comment:
That replacement does seem more straightforward than what's going on here.

But I think the failure is not due to anything about "only affects the current file", it's because `useSelect` comes from `@wordpress/data` rather than `@wordpress/element`. I think the wording you're looking at from https://jestjs.io/docs/jest-object#jestmockmodulename-factory-options is referring to different _test files_, not to files required by the _current_ test.

I see [extensions/blocks/gathering-tweetstorms/test/editor.js](https://github.com/Automattic/jetpack/blob/74ba77bc25f6cd5879c8996e6c82b46e0bc39c2d/projects/plugins/jetpack/extensions/blocks/gathering-tweetstorms/test/editor.js) successfully mocks `useSelect` by doing it in a different way, for example.

Reply:
0625b7cbaab516dbdeae1ee1d9b47dac5a24fe9f did it. Thank you both!